lib/broadway_rabbitmq/producer.ex

```diff
@@ -97,31 +100,75 @@ defmodule BroadwayRabbitMQ.Producer do
               host: "192.168.0.10"
             ],
             qos: [
-              prefetch_count: 50
+              # See "Back-pressure and `:prefetch_count`" section
+              prefetch_count: @processor_concurrency * @max_demand
             ]},
+          # See "Producer concurrency" section
           concurrency: 1
         ],
         processors: [
-          default: []
+          default: [
+            concurrency: @processor_concurrency,
+            max_demand: @max_demand
+          ]
         ]
       )
 
+  ## Producer concurrency
+
+  For efficiency, you should generally limit the amount of internal queueing.
+  Whenever additional messages are sitting in a busy processor's mailbox, they
+  can't be delivered to another processor which may be available or become
+  available first.
+
+  One posible cause of internal queueing is multiple producers. This is because
+  each processor's demand will be sent to all producers. For example, if a
+  processor demands `2` messages and there are `2` producers, each producer
+  will try to  pull `2` messages and give them to the processor. So the
+  processor may receive `max_demand * <producer concurrency>` messages.
+
+  Setting `max_demand` to `1` and producer `concurrency: 1` will eliminate
+  internal queueing, so these are the recommended settings to start with.
```

Comment:
I am not sure this is correct advice. Having max_demand of 1 means there is waiting time after the element has been processed. For elements that are very fast to process, this will show up.

Reply:
The other thing is that consuming one by one increases the number of messages flowing through the topology too. Perhaps I would say: if your messages take time to process (for example, you need to do multiple queries in the database), then consider setting max_demand to 1. If they are all fast, the default number of 10 can be a good starting point.

Reply:
Gotcha. Revised.

Reply:
Thanks. See https://github.com/dashbitco/broadway_rabbitmq/pull/94